doc/how-to/vshard_quick.rst

```diff
@@ -1,160 +1,525 @@
+    -   ``storage.lua`` is intended to store code specific for :ref:`storages <vshard-architecture-storage>`.
+    -   ``router.lua`` is intended to store code specific for a :ref:`router <vshard-architecture-router>`.
+    -   ``sharded_cluster-scm-1.rockspec`` includes external dependencies required by the application.
+
+    The next :ref:`vshard-quick-start-developing-app` section shows how to configure the cluster and write code specific for a router and storages.
+
+
+..  _vshard-quick-start-developing-app:
+
+Developing the application
+--------------------------
+
+..  _vshard-quick-start-configuring-instances:
+
+Configuring instances to run
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Open the ``instances.yml`` file and add the following content:
+
+..  literalinclude:: /code_snippets/snippets/sharding/instances.enabled/sharded_cluster/instances.yaml
+    :language: yaml
+    :dedent:
+
+This file specifies instances to run in the current environment.
+
+
+..  _vshard-quick-start-configuring-cluster:
+
+Configuring the cluster
+~~~~~~~~~~~~~~~~~~~~~~~
+
+This section describes how to configure the cluster in the ``config.yaml`` file.
+
+..  _vshard-quick-start-configuring-cluster-credentials:
+
+Step 1: Configuring credentials
+*******************************
+
+Add the :ref:`credentials <configuration_reference_credentials>` configuration section:
+
+..  literalinclude:: /code_snippets/snippets/sharding/instances.enabled/sharded_cluster/config.yaml
+    :language: yaml
+    :start-at: credentials:
+    :end-at: roles: [sharding]
+    :dedent:
+
+In this section, two users with the specified passwords are created:
+
+*   The ``replicator`` user with the ``replication`` role.
+*   The ``storage`` user with the ``sharding`` role.
+
+..  WARNING::
+
+    It is recommended to load passwords from safe storage such as external files or environment variables.
+    You can learn how to do this from :ref:`configuration_credentials_loading_secrets`.
+
+
+
+..  _vshard-quick-start-configuring-cluster-advertise:
+
+Step 2: Specifying advertise URIs
+*********************************
+
+Add the :ref:`iproto.advertise <configuration_reference_iproto_advertise>` section:
+
+..  literalinclude:: /code_snippets/snippets/sharding/instances.enabled/sharded_cluster/config.yaml
+    :language: yaml
+    :start-after: roles: [sharding]
+    :end-at: login: storage
+    :dedent:
+
+In this section, the following options are configured:
+
+*   ``iproto.advertise.peer`` specifies how to advertise the current instance to other cluster members.
+    In particular, this option informs other replica set members that the ``replicator`` user should be used to connect to the current instance.
+*   ``iproto.advertise.sharding`` specifies how to advertise the current instance to a router and rebalancer.
+
+
+..  _vshard-quick-start-configuring-cluster-bucket-count:
+
+Step 3: Configuring bucket count
```

Comment:
Shoudn't we introduce buckets somehow or provide a link? They just come out if the blue, never introduced before.

Reply:
Added a link to https://www.tarantool.io/en/doc/latest/concepts/sharding/vshard_architecture/#virtual-buckets